flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -454,22 +454,34 @@ public static void checkAcidTable(CatalogTable catalogTable, ObjectPath tablePat
      */
     public static Configuration getHadoopConfiguration(String hadoopConfDir) {
         if (new File(hadoopConfDir).exists()) {
-            Configuration hadoopConfiguration = new Configuration();
+            Configuration hadoopConfiguration = null;
             File coreSite = new File(hadoopConfDir, "core-site.xml");
             if (coreSite.exists()) {
+                if (hadoopConfiguration == null) {
```

Comment:
Instead of checking this for each conf file, we can create a list to remember all the existing files. If the list is not empty, create the configuration and add the resources. Otherwise return null.

Reply:
Okay, let me change it.
I also thought about this, but what I thought at the time was to move the code as little as possible and maintain the original structure